control-plane/catalog/to-consul/resource.go

```diff
@@ -999,3 +1025,26 @@ func (t *ServiceResource) isIngressService(key string) bool {
 func consulHealthCheckID(k8sNS string, serviceID string) string {
 	return fmt.Sprintf("%s/%s", k8sNS, serviceID)
 }
+
+// Sets the passing service weight
+func setServiceWeight(weight string, appsCount int, r *consulapi.CatalogRegistration) error {
```

Comment:
id change this method to return the service weight or an error, providing the value of the annotation as an input. we should avoid sending in the consul registration into it. it will make this is a cleaner interface that is significantly easier to test.

Reply:
Implemented as suggested 😃